src/xlsxwriter.cc

```diff
@@ -12,24 +12,94 @@ Napi::Object Init(Napi::Env env, Napi::Object exports) {
+        Napi::Number::New(env, LXW_VALIDATION_TYPE_DECIMAL);
+    validationTypes["DECIMAL_FORMULA"] =
+        Napi::Number::New(env, LXW_VALIDATION_TYPE_DECIMAL_FORMULA);
+    validationTypes["LIST"] = Napi::Number::New(env, LXW_VALIDATION_TYPE_LIST);
+    validationTypes["LIST_FORMULA"] =
+        Napi::Number::New(env, LXW_VALIDATION_TYPE_LIST_FORMULA);
+    validationTypes["DATE"] = Napi::Number::New(env, LXW_VALIDATION_TYPE_DATE);
+    validationTypes["DATE_FORMULA"] =
+        Napi::Number::New(env, LXW_VALIDATION_TYPE_DATE_FORMULA);
+    validationTypes["DATE_NUMBER"] =
+        Napi::Number::New(env, LXW_VALIDATION_TYPE_DATE_NUMBER);
+    validationTypes["TIME"] = Napi::Number::New(env, LXW_VALIDATION_TYPE_TIME);
+    validationTypes["TIME_FORMULA"] =
+        Napi::Number::New(env, LXW_VALIDATION_TYPE_TIME_FORMULA);
+    validationTypes["TIME_NUMBER"] =
+        Napi::Number::New(env, LXW_VALIDATION_TYPE_TIME_NUMBER);
+    validationTypes["LENGTH"] =
+        Napi::Number::New(env, LXW_VALIDATION_TYPE_LENGTH);
+    validationTypes["LENGTH_FORMULA"] =
+        Napi::Number::New(env, LXW_VALIDATION_TYPE_LENGTH_FORMULA);
+    validationTypes["CUSTOM_FORMULA"] =
+        Napi::Number::New(env, LXW_VALIDATION_TYPE_CUSTOM_FORMULA);
+    validationTypes["ANY"] = Napi::Number::New(env, LXW_VALIDATION_TYPE_ANY);
+    exports["ValidationType"] = validationTypes;
+  }
+
+  {
+    auto validationCriteria = Napi::Object::New(env);
+    validationCriteria["NONE"] =
+        Napi::Number::New(env, LXW_VALIDATION_CRITERIA_NONE);
+    validationCriteria["BETWEEN"] =
+        Napi::Number::New(env, LXW_VALIDATION_CRITERIA_BETWEEN);
+    validationCriteria["NOT_BETWEEN"] =
+        Napi::Number::New(env, LXW_VALIDATION_CRITERIA_NOT_BETWEEN);
+    validationCriteria["EQUAL_TO"] =
+        Napi::Number::New(env, LXW_VALIDATION_CRITERIA_EQUAL_TO);
+    validationCriteria["NOT_EQUAL_TO"] =
+        Napi::Number::New(env, LXW_VALIDATION_CRITERIA_NOT_EQUAL_TO);
+    validationCriteria["GREATER_THAN"] =
+        Napi::Number::New(env, LXW_VALIDATION_CRITERIA_GREATER_THAN);
+    validationCriteria["LESS_THAN"] =
+        Napi::Number::New(env, LXW_VALIDATION_CRITERIA_LESS_THAN);
+    validationCriteria["GREATER_THAN_OR_EQUAL_TO"] = Napi::Number::New(
+        env, LXW_VALIDATION_CRITERIA_GREATER_THAN_OR_EQUAL_TO);
+    validationCriteria["LESS_THAN_OR_EQUAL_TO"] =
+        Napi::Number::New(env, LXW_VALIDATION_CRITERIA_LESS_THAN_OR_EQUAL_TO);
+    exports["ValidationCriteria"] = validationCriteria;
+  }
+
+  {
+    auto validationErrorTypes = Napi::Object::New(env);
+    validationErrorTypes["STOP"] =
+        Napi::Number::New(env, LXW_VALIDATION_ERROR_TYPE_STOP);
+    validationErrorTypes["WARNING"] =
+        Napi::Number::New(env, LXW_VALIDATION_ERROR_TYPE_WARNING);
+    validationErrorTypes["INFORMATION"] =
+        Napi::Number::New(env, LXW_VALIDATION_ERROR_TYPE_INFORMATION);
+    exports["ValidationErrorType"] = validationErrorTypes;
+  }
```

Comment:
Move these to be inside the `DataValidation` class, like `Format`, and add the `_VALIDATION_TYPE`, `_VALIDATION_CRITERIA`, and `_VALIDATION_ERROR_TYPE` suffixes.

Reply:
I've done this, the only problem is now `Worksheet` has to be exported by value, not just by type, so that enum values are accessible from js. Is that fine?

Reply:
Yeah, you can move it to just before `Workbook`. The constants also need to be marked `static readonly` and use `:` not `=` (like the constants in `Format`).